source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1408,6 +1408,18 @@ void ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::drainConnP
   }
 }
 
+// TODO(junr03): clean up this usage and the usage above once ConnPoolImplBase::startDrain and
+// ConnPoolImplBase::drainConnections() get cleaned up. The code in onHostHealthFailure and the code
+// in ThreadLocalClusterManagerImpl::drainConnPools(const HostVector& hosts) is very similar and
+// can be merged in a similar fashion to the ConnPoolImplBase case.
```

Comment:
I think you can just factor out the relevant part of both functions into a helper function that has a better name? I don't think we really need the TODOs here. We can just fix it later if needed.

Reply:
alright. let me do that.